### PR TITLE
aggiunto il futuro tutta implementazione

### PR DESCRIPTION
**Pull request by pre63**

### CODE REVIEW

File: src/future/index.js [hunk 1]
```diff
@@ -27,4 +27,25 @@ Future.of = x => Future((reject, resolve) => resolve(x))
...
+          result =>
+            ((results[i] = result), !errored &&
+              countSparse(results) === futures.length &&
+              right(results))
+        ), results),
+      []
+    ))
+
+// all :: [Future] -> Future
+Future.all = (...futures) => all([].concat(...futures))
```
Comment (gumarm) [inline src/future/index.js]: `Future.all = (...futures) => all(futures)` seems more concise.

File: src/future/index.js [hunk 2]
```diff
@@ -27,4 +27,25 @@ Future.of = x => Future((reject, resolve) => resolve(x))
...
+// countSparse :: Array -> Number
+const countSparse = arr => arr.filter(x => x !== undefined).length
+
+// all :: [Future] -> Future
+const all = futures =>
+  Future((left, right, errored = false) =>
+    futures.reduce(
+      (results, future, i) =>
+        (future.fork(
+          error => !errored && ((errored = true), left(error)),
```
Comment (gumarm) [inline src/future/index.js]: You and your love of that obscure operator hehe

File: src/future/index.js (hunk 1 above)
Comment (oleduc) [inline src/future/index.js]:
```
> const a = (...x) => [].concat(...x)
undefined
> const b = (...x) => x
undefined
> a([1,2])
[ 1, 2 ]
> b([1,2])
[ [ 1, 2 ] ]
```

Not quite the same thing.

Comment (oleduc) [inline src/future/index.js]: Unless I missed something 

```
> const a = (...x) => [].concat(...x)
undefined
> const b = (...x) => x
undefined
> a([1,2])
[ 1, 2 ]
> b([1,2])
[ [ 1, 2 ] ]
```

Comment (gumarm) [inline src/future/index.js]: Yeah, did not think we wanted to support arrays as params! Had a talk with Simon on friday :)

**Pull request merged**